provider/cluster_rosa_classic_resource.go

```diff
@@ -1381,7 +1403,7 @@ func (r *ClusterRosaClassicResource) upgradeClusterIfNeeded(ctx context.Context,
 		return err
 	}
 
-	if !correctUpgradePending {
+	if !correctUpgradePending && !cancelingUpgradeOnly {
 		// Gate agreements are checked when the upgrade is scheduled, resulting
```

Comment:
Also I'm prefer to refactor that logic and move it to another function (lines 1407- 1460) if it possible